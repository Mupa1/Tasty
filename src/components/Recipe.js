import React from 'react';
import PropTypes from 'prop-types';

const Recipe = ({ recipe }) => {
  const { title, image } = recipe;

  return (
    <div>
      <img src={image} alt={title} />
      <h3>{title}</h3>
    </div>
  );
};

Recipe.propTypes = {
  recipe: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    image: PropTypes.string,
  }).isRequired,
};

export default Recipe;
