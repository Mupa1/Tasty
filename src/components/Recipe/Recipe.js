import React from 'react';
import PropTypes from 'prop-types';

import styles from './Recipe.module.css';

const Recipe = ({ recipe }) => {
  const { title, image } = recipe;

  return (
    <div className={styles.recipeContainer}>
      <div className={styles.imgContainer}>
        <img src={image} alt={title} />
      </div>
      <h3>{title}</h3>
    </div>
  );
};

Recipe.propTypes = {
  recipe: PropTypes.shape({
    id: PropTypes.number,
    title: PropTypes.string,
    image: PropTypes.string,
  }).isRequired,
};

export default Recipe;
