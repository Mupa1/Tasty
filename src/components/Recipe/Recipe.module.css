img {
  width: 100%;
  min-height: 100%;
}

.recipeContainer {
  max-width: 300px;
  padding: 10px;
  margin-bottom: 20px;
}

.recipeContainer:hover {
  box-shadow: 0 10px 6px -6px #777;
  max-width: 302px;
}

.imgContainer {
  min-height: 207px;
  max-width: 300px;
}

h3 {
  font-size: 22px;
  color: #555;
}
