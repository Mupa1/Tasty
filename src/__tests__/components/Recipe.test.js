import React from 'react';
import { render, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Recipe from '../../components/Recipe/Recipe';

afterEach(cleanup);

const recipe = {
  title: 'Sample Title',
  image: 'sample.jpg',
};

describe('Recipe', () => {
  it('matches Recipe snapshot', () => {
    const recipes = renderer.create(<Recipe recipe={recipe} />);
    const tree = recipes.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a title', () => {
    const { queryAllByTestId } = render(
      <Recipe recipe={recipe} />,
    );
    const title = queryAllByTestId('title');
    expect(title[0]).toBeTruthy();
  });

  it('renders an image', () => {
    const { queryAllByTestId } = render(
      <Recipe recipe={recipe} />,
    );
    const image = queryAllByTestId('image');
    expect(image[0]).toBeTruthy();
  });

  it('renders a title correctly', () => {
    const { queryAllByTestId } = render(
      <Recipe recipe={recipe} />,
    );
    const title = queryAllByTestId('title');
    expect(title[0]).not.toBeFalsy();
  });

  it('renders an image correctly', () => {
    const { queryAllByTestId } = render(
      <Recipe recipe={recipe} />,
    );
    const image = queryAllByTestId('image');
    expect(image[0]).not.toBeFalsy();
  });
});
