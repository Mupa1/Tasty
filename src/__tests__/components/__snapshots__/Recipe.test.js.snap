// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe matches Recipe snapshot 1`] = `
<div
  className="recipeContainer"
>
  <div
    className="imgContainer"
  >
    <img
      alt="Sample Title"
      data-testid="image"
      src="sample.jpg"
    />
  </div>
  <h3
    data-testid="title"
  >
    Sample Title
  </h3>
</div>
`;
